howso/feature_residuals.amlg

```diff
@@ -603,153 +599,216 @@
 											; each react tuple
 											(map (lambda (get (current_value) 2)) (current_value 1))
 										)
-								))
+								)
+								(map
+									(lambda
+										(zip
+											;all collisions will increase the count value
+											;e.g. zipping (list small medium small) will create (assoc small 1 medium 1)
+											;then accumulate a 1 for the second 'small' in the list since it's a key collision, resulting in (assoc small 2 medium 1)
+											(lambda (+ 1 (current_value 1)) )
+											;ensure to zip a list of classes by prepending an empty list
+											(append (list) (current_value))
+											;zip all classes with a value of 1 so that all clases in the predicted list are assigned a value of 1
+											1
+										)
+									)
+									predicted_classes_map
+								)
+							)
+						))
+						(unzip feature_residuals_lists nominal_indices)
+					)
+				)
+		))
 
-								;create a sparse confusion matrix for each class
-								(assign (assoc
-									confusion_matrix
-										(map
-											(lambda
-												(zip
-													;all collisions will increase the count value
-													;e.g. zipping (list small medium small) will create (assoc small 1 medium 1)
-													;then accumulate a 1 for the second 'small' in the list since it's a key collision, resulting in (assoc small 2 medium 1)
-													(lambda (+ 1 (current_value 1)) )
-													;ensure to zip a list of classes by prepending an empty list
-													(append (list) (current_value))
-													;zip all classes with a value of 1 so that all clases in the predicted list are assigned a value of 1
-													1
-												)
-											)
-											predicted_classes_map
+		;accuracy is: num correct predictions / all predictions
+		(assign (assoc
+			accuracy_map
+				(map
+					(lambda
+						(let 
+							;grab the current action feature's confusion matrix
+							(assoc current_confusion_matrix (get confusion_matrix_map (current_index 1)))
```

Comment:
since confusion_matrix_map is a map of feature -> confusion_matrix, go ahead and get rid of the 'current_' in front, so it's just 'confusion_matrix'. and do it for the others below as well.   if you feel that it needs something prepended, then perhaps feature_confusion_matrix is more clear.

Reply:
yup, makes sense, was a vestige from some previous variable names